### DISCUSSION TITLE
leantime behind nginx reverse-proxy

### DISCUSSION THREAD

**Discussion by andyhough** [Leantime/leantime#663] [Q&A]
So I am trying to use leantime from a docker image on my Centos 8 vpn.  the container is up and running, but when I go to the URL the reverse proxy returns a 502 Bad Gateway.

I have nginx-proxy running with nginx-proxy-companion to automate the cert creation.  

Has anyone done this?  Can you point me in the right direction?

Thanks!

Below is the (scrubbed) compose file I am loading leantime with.  Still in 'getting it installed and connecting phase"  So nothing here that I am committed to.

version: '3.3'

services:
   db:
     image: mysql:5.7
     container_name: mysql_leantime2
     volumes:
       - db_data:/var/lib/mysql2
     restart: always
     environment:
         MYSQL_ROOT_PASSWORD: '321.qwerty'
         MYSQL_DATABASE: 'leantime'
         MYSQL_USER: 'admin'
         MYSQL_PASSWORD: '321.qwerty'
# NOTE: I have another mysql image so I changed the port to 3037 to add a leantime specific mysql container. 
     ports:
       - "3307:3307"
     command: --character-set-server=utf8 --collation-server=utf8_unicode_ci
   web:
     image: leantime/leantime:latest
# NOTE: Another OOTB image is up and running, with the same results, so I create a new container called leantime2
     container_name: leantime2
     environment:
         LEAN_DB_HOST: 'mysql_leantime2'
         LEAN_DB_USER: 'admin'
         LEAN_DB_PASSWORD: '321.qwerty'
         LEAN_DB_DATABASE: 'leantime'

         VIRTUAL_HOST: leantime.example.com
         LETSENCRYPT_HOST: leantime.example.com
         VIRTUAL_PORT: 9002
#         LETSENCRYPT_EMAIL: =admim@example.com
         NETWORK: proxy

     ports:
       - "9002:9002"
#       - "80:80"
     depends_on:
       - db
volumes:
    db_data: {}

networks:
  default:
    external:
      name: nginx-proxy
#      name: ${NETWORK:-proxy}

**Comment by marcelfolaron** [comment]
I am not all too familiar with the set up but shouldn't the virtual_host
and letsencrypt_host point to your acme_companion container?
I would love to see that work though, since this seems to be a lot easier
to set up.

**Comment by andyhough** [comment]
I am still banging on it.  I am pretty sure, per the acme-companion docs that      VIRTUAL_HOST: and 
      LETSENCRYPT_HOST: need to be the same.  The first tells nginx-proxy what FQDN to listen for and send to port VIRTUAL_PORT.  and the second tells letsencrypt what URL to generate and then check the cert for. I suppose if you are using a wildcard cert it may work to put *.example.com in  LETSENCRYPT_HOST  but I doubt it.  I will try as soon as I get something working. 

Ill post my answer when I find it.
